samples/cpp/image_generation/benchmark_text2image.cpp

```diff
@@ -0,0 +1,106 @@
+        return EXIT_SUCCESS;
+    }
+
+    std::string prompt = result["prompt"].as<std::string>();
+    const std::string models_path = result["model"].as<std::string>();
+    std::string device = result["device"].as<std::string>();
+    size_t num_warmup = result["num_warmup"].as<size_t>();
+    size_t num_iter = result["num_iter"].as<size_t>();
+    const std::string output_dir = result["output_dir"].as<std::string>();
+
+    ov::genai::Text2ImagePipeline pipe(models_path, device);
+    ov::genai::ImageGenerationConfig config = pipe.get_generation_config();
+    config.width = result["width"].as<size_t>();
+    config.height = result["height"].as<size_t>();
+    config.num_inference_steps = result["num_inference_steps"].as<size_t>();
+    config.num_images_per_prompt = result["num_images_per_prompt"].as<size_t>();
+    pipe.set_generation_config(config);
+    
+    for (size_t i = 0; i < num_warmup; i++) {
+        pipe.generate(prompt);
+    }
+
+    std::vector<float> generate_durations;
+    std::vector<float> total_inference_durations;
+    float load_time;
+    for (size_t i = 0; i < num_iter; i++) {
+        ov::Tensor image = pipe.generate(prompt);
+        ov::genai::ImageGenerationPerfMetrics metrics = pipe.get_performance_metrics();
+        generate_durations.emplace_back(metrics.get_generate_duration());
+        total_inference_durations.emplace_back(metrics.get_inference_total_duration());
+        std::string image_name = output_dir + "/image_" + std::to_string(i) + ".bmp";
+        imwrite(image_name, image, true);
+        load_time = metrics.get_load_time();
+    }
+
+    float generate_mean = std::accumulate(generate_durations.begin(),
+                                          generate_durations.end(),
+                                          0.0f,
+                                          [](const float& acc, const float& duration) -> float {
+                                              return acc + duration;
+                                          });
+    if (!generate_durations.empty()) {
+        generate_mean /= generate_durations.size();
+    }
+
+    float inference_mean = std::accumulate(total_inference_durations.begin(),
+                                           total_inference_durations.end(),
+                                           0.0f,
+                                           [](const float& acc, const float& duration) -> float {
+                                               return acc + duration;
+                                           });
+    if (!total_inference_durations.empty()) {
+        inference_mean /= total_inference_durations.size();
+    }
+
+    std::cout << std::fixed << std::setprecision(2);
+    std::cout << "Load time: " << load_time << " ms" << std::endl;
+    std::cout << "One generate avg time: " << generate_mean << " ms" << std::endl;
+    std::cout << "Total inference for one generate avg time: " << inference_mean << " ms" << std::endl;
```

Comment:
can we print more information? E.g. how much time is taken by text encoders, VAE encode / decode, first / other iterations for main denoising loop

Because currently printed information is non informative (about what happens in pipeline) and can be achieved by external benchmarking around generate method.